app/src/main/java/com/stevesoltys/seedvault/ui/storage/RestoreStorageViewModel.kt

```diff
@@ -31,8 +32,6 @@ internal class RestoreStorageViewModel(
                 false
             }
             if (hasBackup) {
-                saveStorage(uri)
-
```

Comment:
First of all, saving the storage off the I/O thread here, possibly on the UI thread is an error that should get resolved in any case. No disk I/O on the UI thread! (If this is happening in other places of the code, that's an error there as well and should get fixed eventually.)

Then, we can't simply move this call around without fully understanding the implications. Just because the backup flow does it, doesn't mean we can do it here as well. If the logic is the same in both flows, why aren't both view models sharing this in common code in `StorageViewModel` then? Just a programming oversight?

I myself don't remember, but looking just at this method here I wonder, what if there are no backups in the location the user has chosen? Will it always be possible to try a different location even if the first one was already saved? Did you try that?

Maybe we don't even need to move the saving call. If we knew what the issue was here, we might be able to come up with a different solution.

Reply:
It looks the only issue that you might have run into is in the initialization of `DocumentsProviderStoragePlugin` which tries to access `storage` at construction time now (which might actually cause issues in other places if storage is not yet available). However, if `storage` is still `null` there, then there shouldn't be any crashes or anything. But it might have happened that you couldn't access the system user USB drive from a work profile at restore time.

Reply:
Moved it back. Will post log when I have it

Reply:
Bump.

@Uldiniad do one thing, split this part into a separate commit - just this one liner.